docs/docs/1.0.6/documentation/health.md

```diff
@@ -0,0 +1,160 @@
+regardless of the environment:
+
+- if proxy status is `Offline`, then Dataplane is `Offline`:
+- if proxy status is `Online`:
+  - if all inbounds are ready then Dataplane is `Online`
+  - if all inbounds are not ready then Dataplane is `Offline`
+  - if at least one of the inbounds is not ready then Dataplane is `Partially degraded` 
+  - if inbound is not ready then it's not included in the load-balancer set which means it doesn't receive the traffic
+  - if all inbounds which implement the same service are ready then service is `Online`
+  - if all inbounds which implement the same service are not ready then service is `Offline`
+  - if at least one of the inbounds which implement the same service is not ready then service is `Partially degraded`
+
+## Kubernetes probes
+
+Kuma natively supports the `httpGet` Kubernetes probes. By default, Kuma overrides the specified probe with a virtual one. For example, if we specify the following probe:
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /metrics
+    port: 3001
+  initialDelaySeconds: 3
+  periodSeconds: 3
+```
+
+Kuma will replace it with:
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /3001/metrics
+    port: 9000
+  initialDelaySeconds: 3
+  periodSeconds: 3
+```
+
+Where `9000` is a default virtual probe port, which can be configured in `kuma-cp.config`:
+
+```yaml
+runtime:
+  kubernetes:
+    injector:
+      virtualProbesPort: 19001
+```
+And can also be overwritten in the Pod's annotations:
+
+```yaml
+annotations:
+  kuma.io/virtual-probes-port: 19001
+```
+
+To disable Kuma's probe virtualziation, we can either set it in Kuma's configuration gile `kuma-cp.config`:
+
+```yaml
+runtime:
+  kubernetes:
+    injector:
+      virtualProbesEnabled: true
+```
+
+or in the Pod's annotations:
+
+```yaml
+annotations:
+  kuma.io/virtual-probes: enabled
+```
+
+:::tip
+Even if virtual probes are disabled Kuma takes `pod.status.containerStatuses.ready` in order to fill `dataplane.inbound.health` section.
+:::
+
+## Universal probes
+
+On Universal there is no single standard for probing the application. For health checking of the application status on
+Universal Kuma is using Envoy's Health Discovery Service (HDS). Envoy does health checks and reports the status back to Kuma Control Plane.
+
+In order to configure health checking of your application you have to update `inbound` config with `serviceProbe`:
+
+```yaml
+type: Dataplane
+mesh: default
+name: web-01
+networking:
+  address: 127.0.0.1
+  inbound:
+    - port: 11011
+      servicePort: 11012
+      serviceProbe:
+        timeout: 2s
+        interval: 1s
+        healthyThreshold: 1
+        unhealthThreshold: 1
+        tcp: {}
+      tags:
+        kuma.io/service: backend
+        kuma.io/protocol: http
+```
+
+`Timeout`, `Interval`, `HealthyThreshold` and `UnhealthyThreshold` are optional. Default values for them are configured in kuma-cp config:
+
```

Comment:
as a first time reader: why `health` is missing?

Present two options to users
1) Use `serviceProbe`, `health.ready` will be updated automatically by the control plane
2) You write the script or whatever that updates `health.ready=true` for you (+ tip remember of starting DP with health.ready=false) and not use `serviceProbe`